metricbeat/module/zookeeper/connection/data.go

```diff
@@ -70,14 +69,14 @@ func (m *MetricSet) parseCons(i io.Reader) ([]mb.Event, error) {
 		}
 	}
 
-	return result, nil
+	return result
 }
 
 func lineToMap(line string) (map[string]string, error) {
 	capturedPatterns := capturer.FindStringSubmatch(line)
 	if len(capturedPatterns) < 1 {
 		//Nothing captured
-		return nil, errors.Errorf("no data captured in '%s'", line)
+		return nil, fmt.Errorf("no data captured in '%s'", line)
```

Comment:
It could be nice if we could identify at some point if this parsing is failing because of disabled commands.

Reply:
Probably it is enough by now to print the error at the error level as you are doing in https://github.com/elastic/beats/pull/31013/files#diff-5f860c45dcb933cac184b249b73f249ae48397636487025ecca625e0b0a00d89R47, this already prints the data returned in this case (`cons is not executed because it is not in the whitelist.`)

To identify if this is an error caused by misconfiguration of the whitelist, we would need to do something like:
* Checking the config, but not sure if this is possible, or if this would require another layer of permissions.
* Parsing the response. But as this seems to be a plain string, it may be error prone, or don't work at all if these strings are changed in future versions.